src/main/java/com/twilio/oai/TwilioTerraformGenerator.java

```diff
@@ -212,43 +259,47 @@ private CodegenOperation getCodegenOperation(final Map<String, Object> resource,
         return Stream.concat(createParams.stream(), updateStream).collect(Collectors.toList());
     }
 
-    private void addSchemaVendorExtensions(final CodegenParameter codegenParameter, final String schemaType) {
-        final String terraformSchemaType = buildSchemaType(codegenParameter, schemaType);
+    private void addSchemaVendorExtensions(final CodegenParameter codegenParameter,
+                                           final TerraformSchemaOptions schemaOptions) {
+        final TerraformSchema terraformSchema = buildTerraformSchema(codegenParameter, schemaOptions);
 
-        codegenParameter.vendorExtensions.put("x-terraform-schema-type", terraformSchemaType);
+        codegenParameter.vendorExtensions.put("x-terraform-schema", terraformSchema);
         codegenParameter.vendorExtensions.put("x-name-in-snake-case", this.toSnakeCase(codegenParameter.baseName));
     }
 
     private Set<String> getParamNames(final List<CodegenParameter> parameters) {
         return parameters.stream().map(param -> param.paramName).collect(Collectors.toSet());
     }
 
-    private String buildSchemaType(final CodegenParameter codegenParameter, final String schemaType) {
-        return buildSchemaType(codegenParameter.dataType, schemaType, codegenParameter.items);
+    private TerraformSchema buildTerraformSchema(final CodegenParameter codegenParameter,
+                                                 final TerraformSchemaOptions schemaOptions) {
+        return buildTerraformSchema(codegenParameter.dataType, schemaOptions, codegenParameter.items);
     }
 
-    private String buildSchemaType(final CodegenProperty codegenProperty, final String schemaType) {
-        return buildSchemaType(codegenProperty.dataType, schemaType, codegenProperty.items);
+    private TerraformSchema buildTerraformSchema(final CodegenProperty codegenProperty,
+                                                 final TerraformSchemaOptions schemaOptions) {
+        return buildTerraformSchema(codegenProperty.dataType, schemaOptions, codegenProperty.items);
     }
 
-    private String buildSchemaType(final String dataType, final String schemaType, final CodegenProperty items) {
+    private TerraformSchema buildTerraformSchema(final String dataType,
+                                                 final TerraformSchemaOptions schemaOptions,
+                                                 final CodegenProperty items) {
         if (dataType.startsWith("[]")) {
-            return String.format("AsList(%s, %s)", buildSchemaType(items, schemaType), schemaType);
+            return new TerraformSchema("list", schemaOptions, buildTerraformSchema(items, schemaOptions));
         }
 
         switch (dataType) {
             case "float32":
```

Comment:
Should we be handling other types such as long(int64)/double(float64), etc. now or wait until we have a spec with those types defined?

Reply:
Out of scope for this PR, but I say wait as it will require changes elsewhere to support (i.e., yps).